config/boards/imx7sabre.csc

```diff
@@ -0,0 +1,10 @@
+# iMX7d dual core + M4 1Gb RAM SoC 1xmPCIe 2xGBE
+BOARD_NAME="MCIMX7SABRE"
+BOARDFAMILY="mx7dsabre"
```

Comment:
better change to more generic name like imx7 ... remember to adjust patch directories accordingly.

Reply:
I can change BOARDFAMILY but not sure it's a good idea because it use specific branches of u-boot and kernel for this particular board family. For different iMX7-based board we would use different branches.

Reply:
u-boot is always board specific and in case you share u-boot source among some boards, you can have board specifics patches ([example](https://github.com/armbian/build/tree/master/patch/u-boot/u-boot-rockchip64-default))

Kernel should (in theory) be agnostic, but I do agree its not simple ... we have two different kernels for RK3399 chip for example ([and desperately wanted to merge them](https://forum.armbian.com/topic/10282-merge-rockchip64-with-rk3399/)) ... maintaining several sources is costly.

Since this is first imx7 board, its easy to set BOARDFAMILY to imx7 and others should either choose something else or merge with existing imx7 legacy kernel ...

It is also possible to have BOARDFAMILY=x and LINUXFAMILY=y Check https://github.com/armbian/build/blob/master/config/sources/sun50iw1.conf#L9

Pine64 = board family is sun50iw1 which is internal code
default legacy kernel name = LINUXFAMILY=pine64
next and dev = LINUXFAMILY=sunxi64